spec/message.abnf

```diff
@@ -1,18 +1,30 @@
-message = pattern / complex-message
+message = simple-message / complex-message
```

Comment:
This makes `{{Pattern}}` a valid complex message, in addition to `Pattern` being a valid simple message.

I think this is good. It's a degenerate case of a complex message with no declarations and one implicit variant. I also think it's consistent with how many storage format that allow unquoted content also allow quoting it.

I'm calling it out to make sure we all agree about this. I recall  `.{{Pattern}}` being brought up earlier, which I think would be worse. By choosing dotted keywords as code introducers we effectively freed up `{{`/`}}`, which now are only used unambiguously for quoting patterns.

Reply:
That was the case before as well, yes? Only with the previous style it would've needed to be quoted as `{{{{Pattern}}}}`.

Reply:
Also noting that this (as before) makes the syntax at least LL2, as we need to look at the second character after the first `{` to determine if it's a placeholder, or a complex-mode pattern quote.

Reply:
@stasm noted:
> I'm calling it out to make sure we all agree about this. I recall .{{Pattern}} being brought up earlier, which I think would be worse. By choosing dotted keywords as code introducers we effectively freed up {{/}}, which now are only used unambiguously for quoting patterns.

The design/choice doc explicitly calls out that the `.keyword` syntax makes `{{Pattern}}` a quoted pattern "complex" message because the dot isn't special--it is part of the keyword. This is a _feature_ of the design 😸

Reply:
Ah, you're right, I must have missed it. Or rather, https://github.com/unicode-org/message-format-wg/pull/521#discussion_r1390250275 made me believe that it wasn't the case. Thanks for confirming!